odl/space/space_utils.py

```diff
@@ -22,14 +22,27 @@
 from future import standard_library
 standard_library.install_aliases()
 
-__all__ = ('vector',)
+__all__ = ('vector', 'ntuples', 'fn', 'cn', 'rn',
+           'NTUPLES_IMPLS', 'FN_IMPLS')
 
 import numpy as np
 
-from odl.space.ntuples import Rn, Cn, Fn, Ntuples
+from odl.set import RealNumbers, ComplexNumbers
+from odl.space.numpy_ntuples import NumpyNtuples, NumpyFn
 from odl.util.utility import (
     is_real_floating_dtype, is_complex_floating_dtype, is_scalar_dtype)
-from odl.space.cu_ntuples import CUDA_AVAILABLE, CudaRn, CudaFn, CudaNtuples
+from pkg_resources import iter_entry_points
+
+
+NTUPLES_IMPLS = {'numpy': NumpyNtuples}
+FN_IMPLS = {'numpy': NumpyFn}
+for entry_point in iter_entry_points(group='odl.space', name=None):
+    try:
+        module = entry_point.load()
+        NTUPLES_IMPLS.update(module.ntuples_impls())
+        FN_IMPLS.update(module.fn_impls())
+    except ImportError:
```

Comment:
One question: does this try to import the "plugin module" that shall dock at the entry point (from the exception that's what I guess)? If yes, will it fail if the module is misconfigured, e.g. lacks a library? 

Another point that came to my mind is that sometimes you successfully import the CUDA stuff, but at runtime it crashes, e.g. due to a `MemoryError` or simply because no GPU can be found. I think it would be good if the module would be excluded in that case, since many tests will fail in that scenario, and it happens basically every time you suspend & resume on Linux (some Nvidia driver or Linux kernel bug). Technically one could implement a very simple import hook which does a minimal test and bails out on failure.